app/Console/Commands/UpdateSignupAndPostField.php

```diff
@@ -0,0 +1,104 @@
+use Rogue\Managers\SignupManager;
+
+class UpdateSignupAndPostField extends Command
+{
+    /**
+     * The name and signature of the console command.
+     *
+     * @var string
+     */
+    protected $signature = 'rogue:updatefield {--target= : The name of the field to update} {--targetOldValue= : The value to search the target with} {--targetNewValue= : The value to update the target with}';
+
+    /**
+     * The console command description.
+     *
+     * @var string
+     */
+    protected $description = 'Change all target old values to target new value in the signups and posts tables';
+
+    /**
+     * The signup manager instance.
+     *
+     * @var Rogue\Managers\SignupManager
+     */
+    protected $signups;
+
+    /**
+     * Create a new command instance.
+     *
+     * @return void
+     */
+    public function __construct(SignupManager $signups, PostManager $posts)
+    {
+        parent::__construct();
+
+        $this->signups = $signups;
+        $this->posts = $posts;
+    }
+
+    /**
+     * Execute the console command.
+     *
+     * @return mixed
+     */
+    public function handle()
+    {
+        // Start updating signups
+        info('rogue:updatefield: Starting to update signups!');
+
+        if (! $this->option('target')) {
+            $this->error('No target field specified.');
+
+            return;
+        }
+
+        if (! $this->option('targetOldValue')) {
+            $this->error('No target old value specified.');
+
+            return;
+        }
+
+        if (! $this->option('targetNewValue')) {
+            $this->error('No target new value specified.');
+
+            return;
+        }
+
+        $targetField = $this->option('target') ?? null;
+        $targetOldValue = $this->option('targetOldValue') && $this->option('targetOldValue') !== 'NULL' ? $this->option('targetOldValue') : null;
+        $targetNewValue = $this->option('targetNewValue') && $this->option('targetNewValue') !== 'NULL' ? $this->option('targetNewValue') : null;
+
+        // Get all signups that have "targetOldValue" set as their target and update to "targetNewValue"
+        Signup::withTrashed()->where($targetField, $targetOldValue)->chunkById(100, function ($signups) use ($targetField, $targetNewValue) {
+            foreach ($signups as $signup) {
+                info('rogue:updatefield: changing target field to new value for signup ' . $signup->id);
+                $this->signups->update($signup, [$targetField => $targetNewValue]);
+            }
+        });
+
+        // Log that updating signups are finished
+        info('rogue:updatefield: Finished updating signups!');
+
+        // Start updating posts
+        info('rogue:updatefield: Starting to update posts!');
+
+        // Get all posts that have "sms-mobilecommons" set as their source and update to "sms"
+        Post::withTrashed()->where($targetField, $targetOldValue)->chunkById(100, function ($posts) use ($targetField, $targetNewValue) {
```

Comment:
What happens if we are updating a field that only lives on `signups` but not on `posts`? Does this error out?

Reply:
@katiecrane that is a good question! Yes - we'd _only_ want to use this if we're updating the exact same `target`, searching for the exact same `targetOldValue` and replacing with the exact same `targetNewValue`. This script works for this particular case but maybe we should just separate them into two different scripts right now (one for signups and one for posts)?

Reply:
@chloealee hmm I wonder if it could make sense to here use an option for which models to deal with? Seems easier to run just one command if you DO want to update something that lives on both. Could have like `--signups --posts` for if both should be update? What do you think, is that overkill?

Reply:
oooh @katiecrane I really like that idea - smart!! I'll work on that too in addition to making the other fields arguments instead of options. Thanks!!